main.py

```diff
@@ -38,7 +40,12 @@ def main(self):
                                 title=f"{match}", page_number=page_number-1, access=access_level
                             )
                             # annotated_pages.add(page_number)
-
+                        if key in document.data:
```

Comment:
The else branch here has incorrect indentation- which throws an error. 
Otherwise this looks ok